aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/options/AWSCognitoAuthResendSignUpCodeOptions.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito.options;
+
+import androidx.annotation.NonNull;
+import androidx.core.util.ObjectsCompat;
+
+import com.amplifyframework.auth.options.AuthResendSignUpCodeOptions;
+import com.amplifyframework.util.Immutable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Cognito extension of resend sign up code options to add the platform specific fields.
+ */
+public final class AWSCognitoAuthResendSignUpCodeOptions extends AuthResendSignUpCodeOptions {
+    private final Map<String, String> metadata;
+
+    /**
+     * Advanced options for resend sign up code.
+     * @param metadata Additional custom attributes to be sent to the service such as information about the client
+     */
+    protected AWSCognitoAuthResendSignUpCodeOptions(
+            Map<String, String> metadata
+    ) {
+        this.metadata = metadata;
+    }
+
+    /**
+     * Get custom attributes to be sent to the service such as information about the client.
+     * @return custom attributes to be sent to the service such as information about the client
+     */
+    @NonNull
+    public Map<String, String> getMetadata() {
+        return metadata;
+    }
+
+    /**
+     * Get a builder object.
+     * @return a builder object.
+     */
+    @NonNull
+    public static CognitoBuilder builder() {
+        return new CognitoBuilder();
+    }
+
+    @Override
+    public int hashCode() {
+        return ObjectsCompat.hash(
```

Comment:
`hashCode`, `equals` should have calls to `super()`, right?

Reply:
I don't think so. AuthResendSignUpCodeOptions is an abstract class and doesn't implement hashCode/equals. This is the pattern used for other similar options. Let me know if I am missing something.